lib/connection.js

```diff
@@ -442,7 +442,7 @@ Connection.prototype.resume = function resume () {
 
 Connection.prototype.keyFromFields = function keyFromFields (fields, options) {
   var res = (typeof options.nestTables) + '/' + options.nestTables + '/' + options.rowsAsArray
-    + options.supportBigNumbers + '/' + options.bigNumberStrings;
+    + options.supportBigNumbers + '/' + options.bigNumberStrings + '/' + typeof options.typeCast;
```

Comment:
any chances this key will face collision when `typeCast` function is changed on runtime. I think we need to generate a hash from `typeCast` function and use that for key field. What do you think ?